File: composeApp/src/androidMain/kotlin/dev/restifo/hide_and_seek/util/EnvironmentUtil.kt
package dev.restifo.hide_and_seek.util

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.os.Bundle

/**
 * Utility class for handling environment variables in Android.
 */
object EnvironmentUtil {
    /**
     * Gets an environment variable from the system environment or from the manifest metadata.
     * 
     * @param context The application context
     * @param name The name of the environment variable
     * @param defaultValue The default value to return if the environment variable is not found
     * @return The value of the environment variable, or the default value if not found
     */
    fun getEnvironmentVariable(context: Context, name: String, defaultValue: String = ""): String {
        // First, try to get from system environment
        val envValue = System.getenv(name)
        if (!envValue.isNullOrEmpty()) {
            return envValue
        }
        
        // If not found, try to get from manifest metadata
        return try {
            val appInfo: ApplicationInfo = context.packageManager.getApplicationInfo(
                context.packageName, PackageManager.GET_META_DATA
            )
            val bundle: Bundle = appInfo.metaData ?: Bundle()
            bundle.getString(name) ?: defaultValue
        } catch (e: Exception) {
            e.printStackTrace()
            defaultValue
        }
    }
}

File: iosApp/iosApp/Util/EnvironmentUtil.swift
import Foundation

/// Utility for handling environment variables in iOS
struct EnvironmentUtil {
    /// Gets an environment variable from the process environment or from Info.plist
    /// - Parameters:
    ///   - name: The name of the environment variable
    ///   - defaultValue: The default value to return if the environment variable is not found
    /// - Returns: The value of the environment variable, or the default value if not found
    static func getEnvironmentVariable(name: String, defaultValue: String = "") -> String {
        // First, try to get from process environment
        if let value = ProcessInfo.processInfo.environment[name], !value.isEmpty {
            return value
        }
        
        // If not found, try to get from Info.plist
        if let value = Bundle.main.object(forInfoDictionaryKey: name) as? String, !value.isEmpty {
            return value
        }
        
        // Return default value if not found
        return defaultValue
    }
}
